arena_functions.py

```diff
@@ -43,34 +43,29 @@ def get_gold() -> int:
         return 0
 
 
-def valid_champ(champ: str) -> str:
+def valid_champ(champ: str, comps : CompsManager) -> str:
     """Matches champion string to a valid champion name string and returns it"""
-    if champ in game_assets.CHAMPIONS:
+    if champ in comps.champions:
         return champ
 
-    return next(
-        (
-            champion
-            for champion in game_assets.CHAMPIONS
-            if SequenceMatcher(a=champion, b=champ).ratio() >= 0.7
-        ),
-        "",
-    )
+    for champion in comps.champions:
+        if SequenceMatcher(a=champion, b=champ).ratio() >= 0.7:
+            return champion
+    return ""
```

Comment:
Respective of the logic I implemented a while ago on the main branch, I think it's best to keep the original version of this
```suggestion
    return next(
        (
            champion
            for champion in comps.champions
            if SequenceMatcher(a=champion, b=champ).ratio() >= 0.7
        ),
        "",
    )
```